### PR TITLE
Pollster code review

### PR DESCRIPTION
**Pull request by agyen**
help review pollster

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -13,17 +13,21 @@ app.use(express.static(__dirname + '/dist/poll-ui'));
...
+
 app.post('/sendmail', function(req,res) {
     const user = req.body;
     mail(user.email);
 });
 
-// app.get('/*', function(req,res) {
-// res.sendFile(path.join(__dirname+'/dist/poll-ui/index.html'));
-// });
+
```
Comment (alexowusu) [inline server.js]: remove unused lines of codes

File: src/app/new-poll/new-poll.component.html [hunk 2]
```diff
@@ -3,31 +3,33 @@
...
                   <input formControlName="questions" type="text" class="form-control" id="questionField" placeholder="Question here...">
                 </div>
               <!-- <button (click)="showoptions()" >Continue</button> -->
               
-              <button>Continue</button>
+              <!-- <button>Continue</button> -->
               
 
-          </h5>
+          <!-- </h5> -->
```
Comment (alexowusu) [inline src/app/new-poll/new-poll.component.html]: remove

File: server.js (hunk 1 above)
Comment (agyen) [reply server.js]: thanks mate

File: src/app/new-poll/new-poll.component.html (hunk 2 above)
Comment (agyen) [reply src/app/new-poll/new-poll.component.html]: noted thanks

File: mailgun.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+}
+ 
+const nodemailerMailgun = nodemailer.createTransport(mg(auth));
+
+async function start(toEmail) {
+  
+nodemailerMailgun.sendMail({
+  from: toEmail.creator_email,
+  to: toEmail.recipient_email, 
+//   cc:'second@domain.com',
```
Comment (alexowusu) [inline mailgun.js]: remove

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+dist: trusty
```
Comment (sammoorhouse) [inline .travis.yml]: where's your travis build, I can't see it in the PR page

File: env-playground/server.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+const port = process.env.PORT;
```
Comment (sammoorhouse) [inline env-playground/server.js]: can probably remove this env-playground stuff unless you're using it

File: karma.conf.js [hunk 6]
```diff
@@ -21,6 +21,12 @@ module.exports = function (config) {
       fixWebpackSourcePaths: true
```
Comment (sammoorhouse) [inline karma.conf.js]: if you don't have any UI tests all this stuff just adds to your build & deploy time so consider removing it

File: mailgun.js [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+              <td style="font-family: sans-serif; font-size: 14px; vertical-align: top;">
+              <p style="font-family: sans-serif; font-size: 14px; font-weight: normal; margin: 0; Margin-bottom: 15px;">Hi there,</p>
+              <p style="font-family: sans-serif; font-size: 14px; font-weight: normal; margin: 0; Margin-bottom: 15px;">`+ toEmail.question+`</p>
+              <table border="0" cellpadding="0" cellspacing="0" class="btn btn-primary" style="border-collapse: separate; mso-table-lspace: 0pt; mso-table-rspace: 0pt; width: 100%; box-sizing: border-box;">
+                <tbody>
+                <tr>
+                  <td align="center" style="font-family: sans-serif; font-size: 14px; vertical-align: top; padding-bottom: 15px;">
+                  <table border="0" cellpadding="0" cellspacing="0" style="border-collapse: separate; mso-table-lspace: 0pt; mso-table-rspace: 0pt; width: auto;">
+                    <tbody>
+                    `+ toEmail.options.map(options => (
```
Comment (sammoorhouse) [inline mailgun.js]: nice :)

File: package.json [hunk 8]
```diff
@@ -3,34 +3,52 @@
   "version": "0.0.0",
   "scripts": {
     "ng": "ng",
-    "start": "ng serve",
+    "start": "node server.js",
     "build": "ng build",
     "test": "ng test",
```
Comment (sammoorhouse) [inline package.json]: can you find a way to trim these dependencies? I bet you're not using most of them.

File: src/app/app-routing.module.ts [hunk 9]
```diff
@@ -1,11 +1,22 @@
...
+import {PollManagementComponent } from './poll-management/poll-management.component';
+import { ModelComponent } from './model/model.component';
+import {ThankyoupageComponent} from './thankyoupage/thankyoupage.component'
 
-
-const routes: Routes = [];
+const routes: Routes = [
+	{path: 'matpollform', component: MatpollformComponent},
+	{ path: 'manage', component: PollManagementComponent },
+	{ path: '',   component: ModelComponent},
```
Comment (sammoorhouse) [inline src/app/app-routing.module.ts]: tidy up this whitespace a bit

File: src/app/app.module.ts [hunk 10]
```diff
@@ -1,18 +1,68 @@
 import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
-
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+import { HttpClientModule } from '@angular/common/http';
+import { FusionChartsModule } from 'angular-fusioncharts';
+import * as FusionCharts from 'fusioncharts';
```
Comment (sammoorhouse) [inline src/app/app.module.ts]: where are you using fusioncharts? Kill this stuff until we need it

File: src/app/appservice.service.ts [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    // option.forEach(e =>)
+    return this.http.post<Options>(this._option_url, option);
+    
+  }
+
+  addNewPoll(poll:Poll): Observable<Poll>{
+    return this.http.post<Poll>(this._url, poll);
+  }
+
+  deletePoll(id: number): Observable<Poll>{
```
Comment (sammoorhouse) [inline src/app/appservice.service.ts]: remove these unused options until we need them

File: src/app/matpollform/matpollform.component.ts [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+  styleUrls: ['./matpollform.component.scss']
+})
+
+  export class MatpollformComponent implements OnInit {
+
+  creatorsForm = new FormGroup({
+    email: new FormControl('')
+    });
+
+    user = { email: 'isaac.agyen@turntabl.io'}
```
Comment (sammoorhouse) [inline src/app/matpollform/matpollform.component.ts]: about time to fix this I think :)

File: src/app/matpollform/matpollform.component.ts [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+      selectMultiple: false,
+    };
+  }
+
+
+  ngOnInit() {}
+
+  onSubmit(){
+    this.user.email = this.creatorsForm.value.email;
+    console.log(this.user.email);}
```
Comment (sammoorhouse) [inline src/app/matpollform/matpollform.component.ts]: let's clear out these console.logs please

File: src/app/matpollform/matpollform.component.ts [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+    this.poll.options.splice(index, 1);
+  }
+
+  send() {
+    this.appservice.addNewPoll(this.poll)
+    .subscribe(response => {
+        console.log(response);
+    });
+
+  this.sendmailService.sendmail(this.poll).subscribe()
```
Comment (sammoorhouse) [inline src/app/matpollform/matpollform.component.ts]: watch your indentation here - is this part of the send() function?

File: src/app/matpollform/matpollform.component.ts [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+  this.sendmailService.sendmail(this.creatorsForm.value).subscribe()
```
Comment (sammoorhouse) [inline src/app/matpollform/matpollform.component.ts]: what email is this sending & to whom?

File: src/app/matpollform/sendmail.service.ts [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+import { Observable } from 'rxjs';
+import { User } from '../../app/model/poll'
+
+@Injectable({
+	providedIn: 'root'
+})
+export class SendmailService {
+	constructor(private http: HttpClient) { }
+
+	private mailurl = 'https://polls-angular-ui.herokuapp.com/sendmail'
```
Comment (sammoorhouse) [inline src/app/matpollform/sendmail.service.ts]: can you extract this into an env variable?

File: src/app/model/model.component.html [hunk 17]
```diff
@@ -0,0 +1,8 @@
+<mat-card class="container">
+  <p>
+    Welcome to Turntabl's Poll System.<br /><br />
+    Easily create polls and share them with anyone!<br /><br />
+    Yay! 👏
```
Comment (sammoorhouse) [inline src/app/model/model.component.html]: 😃

File: src/app/poll-management/poll-management.component.html [hunk 18]
```diff
@@ -0,0 +1,22 @@
+<ng-container>
```
Comment (sammoorhouse) [inline src/app/poll-management/poll-management.component.html]: let's remove this until we need it

**Pull request merged**